sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -216,6 +216,35 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
+  val DYNAMIC_PARTITION_PRUNING_ENABLED =
+    buildConf("spark.sql.optimizer.dynamicPartitionPruning.enabled")
+      .doc("When true, we will generate predicate for partition column when it's used as join key")
+      .booleanConf
+      .createWithDefault(true)
+
+  val DYNAMIC_PARTITION_PRUNING_WITH_STATS =
+    buildConf("spark.sql.optimizer.dynamicPartitionPruningWithStats")
```

Comment:
shall we put all the related configs in the same namespace? e.g. `spark.sql.optimizer.dynamicPartitionPruning.withStats`, `spark.sql.optimizer.dynamicPartitionPruning.joinFilterRatio`.